packages/js/tests/dashboard/components/site-kit-setup-widget.test.js

```diff
@@ -0,0 +1,56 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { SiteKitSetupWidget } from "../../../src/dashboard/components/site-kit-setup-widget";
+
+describe( "SiteKitSetupWidget", () => {
+	const defaultProps = {
+		installUrl: "https://example.com/install",
+		activateUrl: "https://example.com/activate",
+		setupUrl: "https://example.com/isSetup",
+		isConnected: false,
+		isActive: false,
+		isSetupCompleted: false,
+		isInstalled: false,
+		onRemove: jest.fn(),
+		onRemovePermanently: jest.fn(),
+		learnMoreLink: "https://example.com/learn-more",
+	};
+
+	it( "renders the widget with install button", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } /> );
+		expect( screen.getByText( "Install Site Kit by Google" ) ).toBeInTheDocument();
+	} );
+
+	it( "renders the widget with activate button", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } isInstalled={ true } /> );
+		expect( screen.getByText( "Activate Site Kit by Google" ) ).toBeInTheDocument();
+	} );
+
+	it( "renders the widget with setup button", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } isInstalled={ true } isActive={ true } /> );
+		expect( screen.getByText( "Set up Site Kit by Google" ) ).toBeInTheDocument();
+	} );
+
+	it( "renders the widget with connect button", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } isInstalled={ true } isActive={ true } isSetupCompleted={ true } /> );
+		expect( screen.getByText( "Connect Site Kit by Google" ) ).toBeInTheDocument();
+	} );
+
+	it( "renders the widget with dismiss button when connected", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } isInstalled={ true } isActive={ true } isSetupCompleted={ true } isConnected={ true } /> );
+		expect( screen.getByText( "Dismiss" ) ).toBeInTheDocument();
+	} );
+
+	it( "opens the menu and calls onRemove when 'Remove until next visit' is clicked", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } /> );
+		fireEvent.click( screen.getByRole( "button", { name: /open menu/i } ) );
+		fireEvent.click( screen.getByText( "Remove until next visit" ) );
+		expect( defaultProps.onRemove ).toHaveBeenCalled();
+	} );
+
+	it( "opens the menu and calls onRemovePermanently when 'Remove permanently' is clicked", () => {
+		render( <SiteKitSetupWidget { ...defaultProps } /> );
+		fireEvent.click( screen.getByRole( "button", { name: /open menu/i } ) );
+		fireEvent.click( screen.getByText( "Remove permanently" ) );
```

Comment:
Is this not a role button? 🤔

Reply:
It is but because it has an svg so I can't get tit by the label

Reply:
The role is menuitem and the type is menu. fixed it.